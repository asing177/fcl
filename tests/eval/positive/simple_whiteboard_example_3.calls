split()
t1()
t2()
join()
