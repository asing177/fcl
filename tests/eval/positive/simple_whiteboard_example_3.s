@{lhsOut, rhsOut}
join() {
  transitionTo(@terminal);
}

@initial
split() {
  transitionTo(@{lhsIn, rhsIn});
}

@lhsIn
t1() {
  transitionTo(@lhsOut);
}

@rhsIn
t2() {
  if (true) {
    if (true) {
      transitionTo(@rhsOut);
    } else {
      transitionTo(@rhsIn);
    };
  } else {
    transitionTo(@rhsOut);
  };
}
