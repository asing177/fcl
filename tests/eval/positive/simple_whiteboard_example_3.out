*** Deltas ***
[ ModifyState
    WorkflowState
      { places =
          fromList
            [ Place { placeName = Name { unName = "lhsIn" } }
            , Place { placeName = Name { unName = "rhsIn" } }
            ]
      }
, ModifyState
    WorkflowState
      { places =
          fromList
            [ Place { placeName = Name { unName = "lhsOut" } }
            , Place { placeName = Name { unName = "rhsIn" } }
            ]
      }
, ModifyState
    WorkflowState
      { places =
          fromList
            [ Place { placeName = Name { unName = "lhsOut" } }
            , Place { placeName = Name { unName = "rhsOut" } }
            ]
      }
, ModifyState WorkflowState { places = fromList [ PlaceEnd ] }
]

*** Globals ***
fromList
  [ ( Key { unKey = "testAddr" }
    , VAccount (Address "fwBVDsVh8SYQy98CzYpNPcbyTRczVUZ96HszhNRB8Ve")
    )
  , ( Key { unKey = "testAddr2" }
    , VAccount (Address "Fw68hsis6JPW4ed3ER1VC959TTf9DqberLxrq5uECKxh")
    )
  , ( Key { unKey = "testAddr3" }
    , VAccount (Address "4Stv5gWGt78D3p919HFcKEschSmyGKGrAX2G35xJgdhw")
    )
  ]
